event.c

```diff
@@ -2314,7 +2314,7 @@ event_priority_set(struct event *ev, int pri)
 
 	if (ev->ev_flags & EVLIST_ACTIVE)
 		return (-1);
-	if (pri < 0 || pri >= ev->ev_base->nactivequeues)
+	if (pri < 0 || pri >= ev->ev_base->nactivequeues || (pri >> (sizeof(ev_uint8_t)*8-1)) != 0)
```

Comment:
`s/ev_uint8_t/ev->ev_pri/` (to avoid problems when `ev_pri` will be expanded)

Reply:
And actually `ev->ev_base->nactivequeues` will never be `>= 256`, take a look at `EVENT_MAX_PRIORITIES` const

Reply:
So I don't think, that we need this bit

Reply:
Plus this check do not accept priority greater then 127 (while it was possible before this patch, see `EVENT_MAX_PRIORITIES`), nack